### ISSUE TITLE
CDAUDIO.WAD Not Implemented ?

### ISSUE THREAD

**Issue by TokyoSU** [opentomb/OpenTomb#557]
I created a cdaudio.wad for tr1 with basic audio loaded for level 1,
but after rewrite the soundtrack load method of tr1 to load wad,
https://github.com/opentomb/OpenTomb/blob/dee5090c0636ef25138a1908ea4a817e2da66b5c/scripts/audio/soundtrack.lua#L720
i have an error in log in-game:
"StreamPlay: CANCEL, wrong track index or broken script."
but all will fine in script and name is correct !

**Comment by vvs-** [comment]
It's not enough to change just that line. You need to change `path` as well _and_ function `return` value too. By default the script uses individual files for each audio track.

**Comment by TokyoSU** [comment]
that have no sense because method are defined to load cdaudio.wad but need to change path and return ?
that break all the function why define wad if the script cant load cdaudio.wad itself ?

**Comment by vvs-** [comment]
The engine itself supports these formats, but apparently there were no efforts to make configuration flexible. I'd remind that OT is of pre-alpha quality with the only developer who put it on hold recently.

**Comment by TokyoSU** [comment]
Sorry, i see that is an alpha and the script (lua) have created but not updated (or just for tr3 new audio),
Please, can you add tag to this issue ?
this function need to be updated for tr3.

**Comment by vvs-** [comment]
There is a temporary solution in https://github.com/opentomb/OpenTomb/issues/347#issuecomment-244590271

**Comment by TokyoSU** [comment]
thanks, but temporary solution give the same error :x
ps: the cdaudio not have all sound into it.

**Comment by TokyoSU** [comment]
hmm need to change how .WAD work for loading, the SHA-1 is really a problem for level creator because if you want more than 128 audio (for big level/game) the SHA-1 cant be a80a34748f400affe0f4a18dc347a7b85e4226e5

**Comment by vvs-** [comment]
No, there is no hardcoded check that files have a particular hash. That was just a simple way for me to verify that your file was genuine, i.e. it should have the same hash if it was from TR3 CD. And if that's different then that particular file might have been corrupted. There is no other way to verify the integrity of some file from unknown source.

OT should work with original .WAD, but it's a moot point right now, because audio support is severely broken in the current version.
